### PR TITLE
Add gradient and time materials

### LINKED ISSUE #7621: Gradient and dt material
### Description of the enhancement or error report
For setting up a feedback phase field simulation that dynamically changes the solute content in a simulation cell to control a particle to a set size I need two small new materials:

1. Norm of the gradient of a variable
2. Current dt

### Rationale for the enhancement or information for reproducing the error

This allows me to set up a `MaskedBodyForce` that shifts an interface laterally

### Identified impact

Additional capability.

### PR DESCRIPTION
**Pull request by dschwen**
Closes #7621

### CODE REVIEW

File: modules/phase_field/src/materials/VariableGradientMaterial.C [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "VariableGradientMaterial.h"
+
+template<>
+InputParameters validParams<VariableGradientMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.addCoupledVar("variable", "Variable to compute the gradinet magnitude of");
```
Comment (laagesen) [inline modules/phase_field/src/materials/VariableGradientMaterial.C]: typo gradinet -> gradient

File: modules/phase_field/tests/misc/timestepmaterial.i [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  kernel_coverage_check = false
+[]
+
+[Executioner]
+  type = Transient
+  num_steps = 5
+
+  [./TimeStepper]
+    type = TimeSequenceStepper
+    time_sequence = '1 4 7 11 56'
```
Comment (laagesen) [inline modules/phase_field/tests/misc/timestepmaterial.i]: How about '4 8 15 16 23 42' instead?

**Comment by laagesen** [comment]
Just 1 typo, otherwise looks good.

**Comment by permcody** [comment]
👍

**Pull request merged**